crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -1341,6 +1341,168 @@ impl<
         self.free_symbol(&Symbol::DEV_TMP3);
     }
 
+    fn build_list_reserve(
+        &mut self,
+        dst: &Symbol,
+        args: &'a [Symbol],
+        arg_layouts: &[InLayout<'a>],
+        ret_layout: &InLayout<'a>,
+    ) {
+        let list = args[0];
+        let list_layout = arg_layouts[0];
+        let spare = args[1];
+        let spare_layout = arg_layouts[1];
+
+        // Load list alignment argument (u32).
+        let u32_layout = Layout::U32;
+        let list_alignment = self.layout_interner.alignment_bytes(list_layout);
+        self.load_literal(
+            &Symbol::DEV_TMP,
+            &u32_layout,
+            &Literal::Int((list_alignment as i128).to_ne_bytes()),
+        );
+
+        // Load element_width argument (usize).
+        let u64_layout = Layout::U64;
+        let element_width = self.layout_interner.stack_size(*ret_layout);
+        self.load_literal(
+            &Symbol::DEV_TMP2,
+            &u64_layout,
+            &Literal::Int((element_width as i128).to_ne_bytes()),
+        );
+
+        // Load UpdateMode.InPlace argument (1u8)
+        let u8_layout = Layout::U8;
+        let update_mode = 1u8;
```

Comment:
Was unsure here, do we want `UpdateMode.InPlace` here or should it be `UpdateMode.Immutable` for this `List.reserve` function?

Reply:
in the dev backend, it will always be immutable. We do not do the analysis there that could ever make it `InPlace`

Reply:
Addressed in e71ea37d3096e5ea0fc0de31ee3ee0881b68882c